test/e2e/es/keystore_test.go

```diff
@@ -105,7 +107,9 @@ func TestUpdateESSecureSettings(t *testing.T) {
 			elasticsearch.CheckESKeystoreEntries(k, b.Elasticsearch, []string{
 				securePasswordSettingKey,
 			}),
-
+			// wait for the rolling upgrade to be fully over
+			elasticsearch.CheckESPodsReady(b, k),
+			elasticsearch.CheckClusterHealth(b, k),
```

Comment:
I thought about factoring that to a single function call; but that function returning a []TestStep would make it hard to use here... so I chose to simply copy/paste the 2 calls here 🤷‍♂

Reply:
Shouldn't we do these checks after updating the secure settings secret?
Have you considered to do these checks in `CheckESKeystoreEntries`?

Reply:
Doing them before is a bit harder since `CheckESPodsReady ` does not take into consideration the secure settings hash, so it would pass with the previous configuration, same for the green health check (would pass before the rolling upgrade starts).
I could patch `CheckESPodsReady  to consider the secure settings hash but not sure it's worth it.
I think I'll go with your suggestion to do the checks within `CheckESKeystoreEntries`. No idea why I did not think about doing that!